gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -280,6 +282,39 @@ public String getCssClass() {
 		};
 		cols.add(studentNameColumn);
 
+		// student extra info optional column
+		String extraStudentProperty = this.businessService.getExtraStudentProperty();
+		if (StringUtils.isNotBlank(extraStudentProperty)) {
+			final AbstractColumn studentExtraInfoColumn = new AbstractColumn(new Model("")) {
+	
+				@Override
+				public Component getHeader(final String componentId) {
+					return new StudentExtraInfoColumnHeaderPanel(componentId, Model.of(settings.getStudentExtraInfoSortOrder()), extraStudentProperty);
+				}
+	
+				@Override
+				public void populateItem(final Item cellItem, final String componentId, final IModel rowModel) {
+					final GbStudentGradeInfo studentGradeInfo = (GbStudentGradeInfo) rowModel.getObject();
+					final Map<String, String> studentProps = studentGradeInfo.getStudentExtraProperties();
+					String studentProperty = "";
+					if (studentProps.containsKey(extraStudentProperty)) {
+						studentProperty = studentProps.get(extraStudentProperty);
+					}
+	
+					final Map<String, Object> modelData = new HashMap<>();
+					modelData.put("extraStudentProperty", studentProperty);
+					cellItem.add(new StudentExtraInfoCellPanel(componentId, Model.ofMap(modelData)));
+				}
+	
+				@Override
+				public String getCssClass() {
+					return "gb-student-extrainfo-cell";
+				}
+	
+			};
+			cols.add(studentExtraInfoColumn);
+		}
+
```

Comment:
@payten is this going the break the frozen scrolling behaviour? Also how does this sit with the table refactor?